src/util/log.cc

```diff
@@ -231,18 +231,20 @@ static internal::log_buf::inserter_iterator print_boot_timestamp(internal::log_b
 static internal::log_buf::inserter_iterator print_real_timestamp(internal::log_buf::inserter_iterator it) {
     struct a_second {
         time_t t;
-        std::string s;
+        std::array<char, 32> static_buf; // big enough to hold '2023-01-14 15:06:33'
+        internal::log_buf buf{static_buf.data(), static_buf.size()};
```

Comment:
Using `log_buf` seemed to be the easiest way around the allocation.

I considered also using it without even a static buffer, since the result would be allocations at most only when the size of the string increases (and not always because of course slack is included in the allocation), and I believe this is a fixed-size string so that would be "just once, for the first log message".